wfl/Database/MongoDBManager.cs

```diff
@@ -63,9 +64,21 @@ private string GenerateCollectionName<T>()
         public async Task<T> GetEntityByIdAsync<T>(string id) where T: IEntity
         {
             IMongoCollection<T> collection = db.GetCollection<T>(GenerateCollectionName<T>());
-            IAsyncCursor<T> task = await collection.FindAsync(e => e.Id == id, null);
+            var filter = Builders<T>.Filter.Eq("Id", id);
+            IAsyncCursor<T> task = await collection.FindAsync(filter, null);
             return await task.FirstOrDefaultAsync();
         }
+        public async Task<IEnumerable<T>> GetEntitiesByDateAsync<T>(DateTime date) where T : IEntity
```

Comment:
Yes, this is retrieving entities by its creation date. I think at the moment, while this code will work for all entities, it is only being used internally to check if a food order has been created on a particular date. While this method specifically checks to see if there are any objects with creation time after midnight of a given date, I think cases where we are filtering by equality of a field to a specified value could be made generic using a search criteria. However, I'm not sure that's necessary yet at this stage since we aren't doing much filtering here.